xline/src/storage/kvstore.rs

```diff
@@ -237,7 +237,46 @@ impl KvStoreInner {
             count: kvs.len().cast(),
             ..RangeResponse::default()
         };
-        response.kvs = kvs;
+        if !req.count_only {
+            match (req.sort_target(), req.sort_order()) {
+                (SortTarget::Key, SortOrder::None) => {}
+                (SortTarget::Key, SortOrder::Ascend) => {
+                    kvs.sort_by(|a, b| a.key.cmp(&b.key));
+                }
+                (SortTarget::Key, SortOrder::Descend) => {
+                    kvs.sort_by(|a, b| b.key.cmp(&a.key));
+                }
+                (SortTarget::Version, SortOrder::Ascend | SortOrder::None) => {
+                    kvs.sort_by(|a, b| a.version.cmp(&b.version));
+                }
+                (SortTarget::Version, SortOrder::Descend) => {
+                    kvs.sort_by(|a, b| b.version.cmp(&a.version));
+                }
+                (SortTarget::Create, SortOrder::Ascend | SortOrder::None) => {
+                    kvs.sort_by(|a, b| a.create_revision.cmp(&b.create_revision));
+                }
+                (SortTarget::Create, SortOrder::Descend) => {
+                    kvs.sort_by(|a, b| b.create_revision.cmp(&a.create_revision));
+                }
+                (SortTarget::Mod, SortOrder::Ascend | SortOrder::None) => {
+                    kvs.sort_by(|a, b| a.mod_revision.cmp(&b.mod_revision));
+                }
+                (SortTarget::Mod, SortOrder::Descend) => {
+                    kvs.sort_by(|a, b| b.mod_revision.cmp(&a.mod_revision));
+                }
+                (SortTarget::Value, SortOrder::Ascend | SortOrder::None) => {
+                    kvs.sort_by(|a, b| a.value.cmp(&b.value));
+                }
+                (SortTarget::Value, SortOrder::Descend) => {
+                    kvs.sort_by(|a, b| b.value.cmp(&a.value));
+                }
+            }
+            response.kvs = if req.limit > 0 && kvs.len() > req.limit.cast() {
```

Comment:
req.limit > 0 && kvs.len() > req.limit.cast()   => (req.limit > 0) && (kvs.len() > req.limit.cast()) to make it clearer.